GQI - Extensions - Logging - More detailed info and code example (#3069)
SkylineCommunications/dataminer-docs: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/GQI_Extensions_Logging.md

---
uid: GQI_Extensions_Logging
---

# Logging

From DataMiner 10.4.5/10.5.0 onwards<!-- RN 39043 -->, [ad hoc data sources](xref:Configuring_an_ad_hoc_data_source_in_a_query) and [custom operators](xref:GQI_Custom_Operator) can log their own messages and exceptions within GQI, complementing the existing [GQI core logging functionality](xref:GQI_Logging) (available from DataMiner 10.4.0/10.4.4).

> ## How to log

Reviewer: To be more in line with our usual phrasing, I'd recommend an -ing form here:

```suggestion
## Enabling logging
```

> ## Prerequisites

Author: I still like the original section title better here. Or you could replace it with your "Logging messages" as well.
Either way, this section contains the main steps I wanted to explain and not prerequisites.

```suggestion
## How to log
```

To enable logging, the GQI extension (either an ad hoc data source or a custom operator) must implement the [IGQIOnInit](xref:GQI_IGQIOnInit) interface.

This building block adds the [OnInit](xref:GQI_IGQIOnInit#oninitoutputargs-oninitoninitinputargs-args) lifecycle method, granting access to the [IGQILogger](xref:GQI_IGQILogger) object through the [Logger property](xref:GQI_OnInitInputArgs#properties) of the [OnInitInputArgs](xref:GQI_OnInitInputArgs) class.

> Throughout its lifespan, the extension uses the [IGQILogger](xref:GQI_IGQILogger) object to log messages and exceptions.

Author: The goal here was to stress how *you, the extension creator*,  can take action to log messages. Phrasing it like this gives the impression that there is more info needed to log your own messages but that's not the case.

```suggestion
Throughout its lifespan, the extension can use the [IGQILogger](xref:GQI_IGQILogger) methods to log messages and exceptions.
```

## Logging messages

To log messages from within an ad hoc data source or custom operator:

1.

1.

1.

> 

Author: I think you might have misunderstood the original "How to log" section a bit? I'm sorry that it wasn't very clear.

The section you now labeled as "Prerequisites" contains in fact the exact steps you need to take in your GQI extension to enable logging. After that, creation a log message is just calling one of the IGQILogger methods as you would in any other program.

> 

Reviewer 2: @skyline-rge, I fear my inexperience with GQI extensions is to blame for this - no worries! Is this PR OK to merge?

> 

Author: @EdithLansens no blame, I'm glad you guys take the time to even try to understand everything :)
PR looks OK to me now.
